File: python/draw_triangles.py
#!/usr/bin/env python3

import gif_factory
import sys

gif = gif_factory.create("test_py.gif", 640, 480)

gif_factory.add_color(gif, 0x000000)
gif_factory.add_color(gif, 0x0000ff)
gif_factory.add_color(gif, 0x00ff00)
gif_factory.add_color(gif, 0xff0000)
gif_factory.add_color(gif, 0xff00ff)
gif_factory.add_color(gif, 0xffff00)
gif_factory.add_color(gif, 0xffffff)
gif_factory.init_end(gif)

triangle = [
  0, -50, 0,
 -50, 50, 0,
  50, 50, 0,
]

rotation = [ 2.2, 1.0, 0.0 ]

gif_factory.clear(gif)
gif_factory.draw_rect(gif, 50, 50, 100, 100, 1)
gif_factory.draw_triangle(gif, triangle, 300, 300, 2)
gif_factory.draw_triangle(gif, triangle, rotation, 300, 100, 10, 3)
gif_factory.write_frame(gif)
gif_factory.finish(gif)

print("finish")

File: samples/draw_triangles.cpp
#include <stdio.h>
#include <stdlib.h>

#include "Kohn3D.h"

int main(int argc, char *argv[])
{
  Kohn3D kohn3d(320, 240, Kohn3D::FORMAT_GIF);

  kohn3d.create("test.gif");
  kohn3d.clear();
  kohn3d.add_color(0x000000);
  kohn3d.add_color(0x0000ff);
  kohn3d.add_color(0x00ff00);
  kohn3d.add_color(0xff0000);
  kohn3d.set_bg_color_index(0);
  kohn3d.set_delay(10);
  kohn3d.set_loop_count(Kohn3D::LOOP_INFINITE);
  kohn3d.init_end();

  Kohn3D::Triangle triangle;
  triangle.x0 = 0;
  triangle.y0 = -50;
  triangle.z0 = 0;
  triangle.x1 = -50;
  triangle.y1 = 50;
  triangle.z1 = 0;
  triangle.x2 = 50;
  triangle.y2 = 60;
  triangle.z2 = 0;

  Kohn3D::Triangle triangle_2;
  triangle_2.x0 = -30;
  triangle_2.y0 = -30;
  triangle_2.z0 = 0;
  triangle_2.x1 = 30;
  triangle_2.y1 = -30;
  triangle_2.z1 = 0;
  triangle_2.x2 = 0;
  triangle_2.y2 = 10;
  triangle_2.z2 = 0;

  Kohn3D::Rotation rotation_0;
  Kohn3D::Rotation rotation_1;

  for (float r = 0; r < 6.18; r += 6.18 / 30)
  {
    rotation_0.rz = r;
    rotation_1.ry = r;

    kohn3d.clear();
    kohn3d.draw_triangle(triangle, rotation_0, 160, 120, 0, 2);
    kohn3d.draw_triangle(triangle, rotation_1, 190, 120, 0, 1);
    kohn3d.draw_triangle(triangle_2, 60, 60, 3);
    kohn3d.write_frame();
  }

  kohn3d.finish();

  return 0;
}

